doc/manual.rst

```diff
@@ -4166,6 +4166,112 @@ For the above `openArray` and `var` rules, whether the expressions refers to
 a location derived from a parameter or global must be visible directly from the
 argument expression, no indirection through locals is allowed.
 
+Delimited Continuations
+-----------------------
+
+For reifying a continuation, that is, turning a conceptual continuation into a
+concrete first-class value (a *continuation instance*), the `suspend`:idx: form
+is used.
+
+A continuation instance is the reified continuation (a procedure) plus the
+saved state needed to run it.
+
+Every available call syntax may be used for `suspend`, including the command
+and dot-call syntax. `suspend` is not a real procedure and does not participate
+in overload resolution, but the name in the callee position may be qualified
+as `system.suspend`.
+
+1. let `T` be the type such that the first argument is of type `typedesc[T]`
+2. let `R` be the return type of the routine where `suspend` is used
+3. let `Ctx` be a unique type such that:
+  * `Ctx is object` evaluates to true
+  * `supportsCopyMem(Ctx)` evaluates to false
+  * the size and alignment of `Ctx` are not queriable in a compile-time context
+  * an instance of `Ctx` is always copyable
+4. let `P` be a type such that:
+  * if `T is void`, then `P` is `proc(e: sink Ctx): R {.nimcall.}`
+  * if `T isnot void`, then `P` is `proc(p: sink T, e: sink Ctx): R {.nimcall.}`
+5. let `cont` refer to the *continuation* (a procedure of type `P`)
+6. let `ctx` refer to the saved context (a value of type `Ctx`)
+7. let `L` refer to the location which is named by the
+   second argument
+8. let `blk` refer to the third argument
+
+Both `T` and `R` must be concrete types.
+
+If `R` is `void`, `blk` must be a statement. If `R` is not `void`, `blk` may
+either be an expression fitting type `R` or a statement.
+
+`blk` is typed within a new scope. The scope starts with containing a single
+symbol with type `(Ctx, P)`, with the name provided by the second argument,
+which must be an identifier.
+
+The `suspend` form must not statically appear:
+* inside a `try` block that has a `finally` clauses
+* in a statement list or expression after a `defer` statement
+* inside a `finally` or `except` clause
+* inside a `defer` block
+* inside a `static` or `const` context
+* as or within the third argument of a `suspend`
+* outside of a routine
+* as part of an `iterator` body
+* as part of a `.tailcall` routine body
+
```

Comment:
I think right now we have restrictions for `method`, `converter` and `macro` too, right?

Reply:
It was just a wild guess, apologies.

Though allowing it on `converter` is very questionable, as no one ever explicitly invokes one. But then again I think converters are weird to begin with.

In general this not an issue.